x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/CompletionIT.java

```diff
@@ -0,0 +1,215 @@
+    // Basic Functionality Tests
+    // ============================================
+
+    public void testCompletionBasic() {
+        var query = String.format(Locale.ROOT, """
+            FROM %s
+            | COMPLETION title WITH { "inference_id": "%s" }
+            | KEEP id, completion
+            | LIMIT 5
+            """, TEST_INDEX, COMPLETION_MODEL_ID);
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "completion"));
+            assertColumnTypes(resp.columns(), List.of("integer", "keyword"));
+
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(lessThanOrEqualTo(5)));
+
+            // Verify completion field is not null
+            for (List<Object> row : values) {
+                assertThat(row.get(1), notNullValue());
+            }
+        }
+    }
+
+    public void testCompletionWithCustomFieldName() {
+        var query = String.format(Locale.ROOT, """
+            FROM %s
+            | COMPLETION generated_text = title WITH { "inference_id": "%s" }
+            | KEEP id, generated_text
+            | LIMIT 3
+            """, TEST_INDEX, COMPLETION_MODEL_ID);
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "generated_text"));
+            assertColumnTypes(resp.columns(), List.of("integer", "keyword"));
+
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(lessThanOrEqualTo(3)));
+        }
+    }
+
+    public void testCompletionConcatenatedPrompt() {
+        var query = String.format(Locale.ROOT, """
+            FROM %s
+            | COMPLETION summary = CONCAT(title, ": ", content) WITH { "inference_id": "%s" }
+            | KEEP id, summary
+            | LIMIT 5
+            """, TEST_INDEX, COMPLETION_MODEL_ID);
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "summary"));
+
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(lessThanOrEqualTo(5)));
+
+            // Verify summary field is not null
+            for (List<Object> row : values) {
+                assertThat(row.get(1), notNullValue());
+            }
+        }
+    }
+
+    // ============================================
+    // Row Limit Tests
+    // ============================================
+
+    public void testCompletionRowLimitEnforcement() {
+        var query = String.format(Locale.ROOT, """
+            FROM %s
+            | COMPLETION title WITH { "inference_id": "%s" }
+            | KEEP id, completion
+            """, TEST_INDEX, COMPLETION_MODEL_ID);
+
+        try (var resp = run(query)) {
+            List<List<Object>> values = getValuesList(resp);
+            // Should be limited by the default row limit (100)
+            assertThat(values.size(), lessThanOrEqualTo(100));
```

Comment:
the index we create always has 6 documents - so we are not really testing the limit enforcement.
let's change the `createAndPopulateTestIndex` so that we create an index with more than 100 documents when we test COMPLETION and more than 1000 when we test RERANK.
and here we should check that we get exactly 100 documents back.